### DISCUSSION TITLE
Push Notification instead of Persistent Notification

### DISCUSSION THREAD

**Discussion by badewanne1234** [Tasshack/dreame-vacuum#124] [General]
I want to send a Push Notification instead of a Persistent Notification as soon as the sidebrush or mainbrush is blocked or if there is another error.

For that, I was thinking to listen to the event `dreame_vacuum_error` and its error message.
Upon that, I can then create a push notification for each individual error message.

Does anyone have the correct syntax including the error messages that are fired? It would prevent me of putting the dreame bot into that state by myself, like throwing a cable under the bot ;)

```
dreame_vacuum_error
    entity_id: Vacuum entity
    error: Error description
    code: Fault code of the error
```
Thank you!

**Comment by feerlessleadr** [comment]
Did you ever figure this out? I'm looking for an example of this myself.

**Reply by badewanne1234** [reply]
Yes I did.
Example:
```
platform: state
entity_id:
  - vacuum.yourvacuumname
attribute: error
to: Route
```

Error codes I figured out...:

- Route
- Drop
- Brush
- Side brush
- Forward suffocate

Drop means that the wheels are suspended.

Maybe there are more error codes but I did not figure out more so far...

**Reply by Tasshack** [reply]
https://github.com/Tasshack/dreame-vacuum/blob/master/custom_components/dreame_vacuum/dreame/types.py#L107
Error codes are there.

**Reply by badewanne1234** [reply]
Thats quite a lot of error codes 😁😁

**Reply by Tasshack** [reply]
Because of that you should use events for triggering automations.
https://github.com/Tasshack/dreame-vacuum/blob/master/docs/events.md#error

**Reply by badewanne1234** [reply]
Does listening to events use less resources compared to state changes or why would you rather listen to events instead of state changes?

**Comment by Tasshack** [comment]
https://community.home-assistant.io/t/send-web-ui-persistent-notifications-to-mobile-devices/274891
